src/Selenium2Library/locators/elementfinder.py

```diff
@@ -151,6 +151,17 @@ def _find_by_default(self, browser, criteria, tag, constraints):
         )
         return self._normalize_result(browser.find_elements_by_xpath(xpath))
 
+    def _get_xpath_constraints(self, constraints):
+        xpath_constraints = []
+        for name in constraints:
+            value = constraints[name]
+            if isinstance(value, list):
+                contraint = "@%s[. = '%s']" % (name, "' or . = '".join(value))
+            else:
+                contraint = "@%s='%s'" % (name, value)
+            xpath_constraints.append(contraint)
+        return xpath_constraints
```

Comment:
This seems to be totally fine, but if there would be `_get_xpath_constraint`, looping could be handled by a list comprehension similarly as earlier:

```python
xpath_constraints = [self._get_xpath_constraint(name, value)
                     for name, value in constraints.items()]
```